projects/keycloak/CertificateUtilsProviderFuzzer.java

```diff
@@ -0,0 +1,110 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+///////////////////////////////////////////////////////////////////////////
+import com.code_intelligence.jazzer.api.FuzzedDataProvider;
+import java.io.ByteArrayInputStream;
+import java.security.KeyPair;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Date;
+import org.keycloak.common.crypto.CertificateUtilsProvider;
+import org.keycloak.common.util.KeyUtils;
+import org.keycloak.crypto.def.BCCertificateUtilsProvider;
+import org.keycloak.crypto.elytron.ElytronCertificateUtils;
+import org.keycloak.crypto.fips.BCFIPSCertificateUtilsProvider;
+
+/**
+ * This fuzzer targets the methods in different
+ * Certificate Utils Provider implementation classes
+ * in the crypto package.
+ */
+public class CertificateUtilsProviderFuzzer {
+  private static CertificateFactory cf;
+  private static BCCertificateUtilsProvider bcuPovider;
+  private static ElytronCertificateUtils ecuProvider;
+  private static BCFIPSCertificateUtilsProvider bfcuProvider;
+
+  public static void fuzzerInitialize() {
+    try {
+      // Initialize certificate factory
+      cf = CertificateFactory.getInstance("X.509");
+
+      // Initialize base provider
+      bcuPovider = new BCCertificateUtilsProvider();
+      ecuProvider = new ElytronCertificateUtils();
+      bfcuProvider = new BCFIPSCertificateUtilsProvider();
+    } catch (CertificateException e) {
+      // Known exception
+    }
+  }
+
+  public static void fuzzerTestOneInput(FuzzedDataProvider data) {
+    // Initialise base certificate related object
+    CertificateUtilsProvider provider = null;
+    X509Certificate cert = null;
+    KeyPair keyPair = null;
+
+    try {
+      // Randomly create a certificate utils provider instance
+      switch (data.consumeInt()) {
+        case 1:
+          provider = bcuPovider;
+          break;
+        case 2:
+          provider = ecuProvider;
+          break;
+        case 3:
+          provider = bfcuProvider;
+          break;
+      }
+
+      // Randomly choose which method to invoke
+      Integer choice = data.consumeInt(1, 4);
+      switch (choice) {
+        case 1:
+          cert = (X509Certificate) cf.generateCertificate(
+              new ByteArrayInputStream(data.consumeBytes(data.remainingBytes() / 2)));
+          keyPair = KeyUtils.generateRsaKeyPair(2048);
```

Comment:
Please move all instances of `generateRsaKeyPair` out of the fuzzer body and reuse. It looks like there is no value from generating a new one every time.